src/adservice/Dockerfile.elastic

```diff
@@ -0,0 +1,28 @@
+FROM eclipse-temurin:21-jdk as builder
+
+WORKDIR /usr/src/app/
+
+COPY ./src/adservice/gradlew* ./src/adservice/settings.gradle* ./src/adservice/build.gradle ./
+COPY ./src/adservice/gradle ./gradle
+
+RUN ./gradlew
+RUN ./gradlew downloadRepos
+
+COPY ./src/adservice/ ./
+COPY ./pb/ ./proto
+RUN ./gradlew installDist -PprotoSourceDir=./proto
+
+# -----------------------------------------------------------------------------
+
+FROM eclipse-temurin:21-jre
+
+ARG version=0.3.2
+WORKDIR /usr/src/app/
+
+COPY --from=builder /usr/src/app/ ./
+ADD --chmod=644 https://repo1.maven.org/maven2/co/elastic/otel/elastic-otel-javaagent/$version/elastic-otel-javaagent-$version.jar /usr/src/app/opentelemetry-javaagent.jar
+ENV JAVA_TOOL_OPTIONS=-javaagent:/usr/src/app/opentelemetry-javaagent.jar
+ENV ELASTIC_OTEL_INFERRED_SPANS_ENABLED=true
```

Comment:
Gathers and includes the Java Agent distribution

Reply:
In the future we might have a docker image you could just copy from, which makes the update process even easier as dependabot can handle it.

Reply:
In the future we might have a docker image you could just copy from, which makes the update process even easier as dependabot can handle it.